### PR TITLE
- add xlnt/1.5.0

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **xlnt/1.5.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
xlnt depends on miniz, utfcpp and a modified version of libstudxml.
miniz should unvendored at least.

**Comment by SSE4** [comment]
> xlnt depends on miniz, utfcpp and a modified version of libstudxml.
> miniz should be unvendored at least.

thanks, somehow I didn't notice that, e.g. didn't spot any `find_package` calls. let's do deps first.

**Comment by hnOsmium0001** [comment]
All dependencies are now in CCI (though their libstudxml is tracking HEAD, while ours is a pretty old version from several years ago; may need to do some additional work)

File: recipes/xlnt/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class XlntConan(ConanFile):
+    name = "xlnt"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/tfussell/xlnt"
+    description = ":bar_chart: Cross-platform user-friendly xlsx library for C++11+"
```
Comment (Croydon) [inline recipes/xlnt/all/conanfile.py:10]:
```suggestion
    description = "Cross-platform user-friendly xlsx library for C++11+"
```

File: recipes/xlnt/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    topics = ("conan", 'xlsx', 'spreadsheet', 'excel', 'cpp', 'c-plus-plus', 'api', 'microsoft', 'read', 'write')
```
Comment (Croydon) [inline recipes/xlnt/all/conanfile.py:11]:
```suggestion
    topics = ("xlsx", "spreadsheet", "excel", "cpp", "c-plus-plus", "api", "microsoft", "read", "write")
```

File: recipes/xlnt/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
```
Comment (Croydon) [inline recipes/xlnt/all/conanfile.py:4]:
```suggestion

required_conan_version = ">=1.33.0"

```

File: recipes/xlnt/all/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.1)
+project(cmake_wrapper)
```
Comment (SSE4) [inline recipes/xlnt/all/CMakeLists.txt:2]:
```suggestion
project(cmake_wrapper)

<html>
<body>
set(CMAKE_C_STANDARD 99)
set(CMAKE_C_STANDARD_REQUIRED ON)
set(CMAKE_C_EXTENSIONS OFF)
```

File: recipes/xlnt/all/test_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package_multi"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (SSE4) [inline recipes/xlnt/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```

**Comment by SSE4** [comment]
seems like we need new release of `libstudxml`, API changed in 1.1:
```
 error C2660: 'xml::serializer::end_element': function does not take 2 arguments
```

**Comment by SpaceIm** [comment]
> 
> 
> seems like we need new release of `libstudxml`, API changed in 1.1:
> 
> ```
>  error C2660: 'xml::serializer::end_element': function does not take 2 arguments
> ```
> 
> see also [tfussell/xlnt#111](https://github.com/tfussell/xlnt/issues/111)
> most likely, we need to package this tag: https://git.codesynthesis.com/cgit/libstudxml/libstudxml/tag/?h=v1.1.0-b.9

Yes I tried more than 1 year ago to create a recipe for xlnt, and it depends on 1.1.0 branch of libstudxml. Unfortunately, there is no available tarball of beta releases of 1.1.0

**Comment by SSE4** [comment]
there are some tarballs, not sure if we can re-use them safely:
https://cppget.org/libstudxml/1.1.0-b.9?f=full
https://stage.build2.org/libstudxml/1.1.0-b.9.20210202082911.e729667b0f34

**Comment by SpaceIm** [comment]
This PR can be closed, xlnt recipe has been added by another PR few months ago.

**Comment by uilianries** [comment]
Closing this PR as it has been superseded

**Pull request closed without merging**